### PR TITLE
Adding the run-jshell script to the bin/

### PR DESCRIPTION
**Pull request by jwgibbs**
Added a bash script called run-shell that will open the jshell interactive session and load the jar files in coatjava/lib to the class-path.  The user does not need to add the jar files as options on the command line or set them as environment variables.

### CODE REVIEW

File: bin/run-jshell
Comment (c-dilks) [inline bin/run-jshell]: Using `"$@"` should allow better argument capture (see #79).
```suggestion
jshell --class-path "$JYPATH" "$@"
```

**Comment by jwgibbs** [comment]
Made the recommended edit.  Switched "$*" with "$@" for the command line inputs.

File: bin/run-jshell
Comment (c-dilks) [reply bin/run-jshell]: Thanks, resolved (not sure why I don't see the "mark as resolved button", but if you have it, please click it, otherwise one of the repo admins will need to).

**Comment by jwgibbs** [comment]
Thanks.

I clicked the Closed with Comment, and now the pull request is marked as Closed.  I don’t know if that did anything.

Mike

/\/\\-\\/\//\/\\-\\/\//\/\\-\\/\//\/\\-\\/\//\/\\-\\/\//\/\\-\\/\//\/\\-\\/\//\/\\-\\/\/
Experimental Nuclear Physicist
Professor/Chair Quantitative Sciences Department
Director Pre-engineering Program
Canisius College
716-888-2426
716-888-2499 (Chair’s Office)
/\/\\-\\/\//\/\\-\\/\//\/\\-\\/\//\/\\-\\/\//\/\\-\\/\//\/\\-\\/\//\/\\-\\/\//\/\\-\\/\/



On Aug 21, 2023, at 3:50 PM, Christopher Dilks ***@***.******@***.***>> wrote:

@c-dilks commented on this pull request.

________________________________

In bin/run-jshell<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_JeffersonLab_coatjava_pull_85-23discussion-5Fr1300576168&d=DwMCaQ&c=CJqEzB1piLOyyvZjb8YUQw&r=-KNeEZI7ShpbkRX5FgA_3aDmyJJkbPigBmPHDUbP87k&m=25AxSdDWHfmyWCxS7N4tGAQYGuMVynBZr55Wlb6OgzMvGDcHD2wBn2LgHOyH5tas&s=Q4iNGaedTT_dZMnOhhPBPcb8qmyUzd8rST_9sgrYONs&e=>:

> +do
+if [ -z "${JYPATH}" ] ; then
+JYPATH="$i"
+else
+JYPATH=${JYPATH}:"$i"
+fi
+done
+#-------------------------------------------------------------
+echo " "
+echo " "
+echo "*****************************************"
+echo "*    Running COAT-JAVA JShell Scripts   *"
+echo "*****************************************"
+echo " "
+echo " "
+jshell --class-path "$JYPATH" $*


Thanks, resolved (not sure why I don't see the "mark as resolved button", but if you have it, please click it, otherwise one of the repo admins will need to).

**Comment by c-dilks** [comment]
> I clicked the Closed with Comment, and now the pull request is marked as Closed. I don’t know if that did anything.

"Closed" means this Pull Request will no longer be considered for merging; you should re-open it (unless you no longer want it); I believe there is a button near the bottom to do this.

**Comment by c-dilks** [comment]
On the other hand, my comments above were part of a "Review", and "Review Conversations" typically need to be "Resolved" before the maintainers can merge a Pull Request. You've added the commit, but there should be a "Resolve" button somewhere near my comments above (if not, the maintainers can do this).

File: bin/run-jshell
Comment (c-dilks) [inline bin/run-jshell:50]: Do you need to `export JAVA_OPTS` too, as `bin/run-groovy` does?

On that note, this script has significant overlap with `bin/run-groovy`, which makes me wonder if they could be combined, e.g., move common things like `JYPATH` into `bin/env.sh`... any thoughts @baltzell?

**Comment by jwgibbs** [comment]
I will admit that my script is a cut and paste of run-groovy.   Maybe we could streamline it by moving the environment variables to env.sh.

Mike

/\/\\-\\/\//\/\\-\\/\//\/\\-\\/\//\/\\-\\/\//\/\\-\\/\/
Professor of Physics
Director of the Pre-Engineering Program
Chair of the Quantitative Sciences Department
Canisius College
716-888-2426
/\/\\-\\/\//\/\\-\\/\//\/\\-\\/\//\/\\-\\/\//\/\\-\\/\/

On Aug 21, 2023, at 4:10 PM, Christopher Dilks ***@***.***> wrote:

﻿
CAUTION: This email was sent from outside of Canisius University.
Do not open attachments or click on links from unknown senders or unexpected emails.



@c-dilks commented on this pull request.

________________________________

In bin/run-jshell<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_JeffersonLab_coatjava_pull_85-23discussion-5Fr1300597413&d=DwMCaQ&c=CJqEzB1piLOyyvZjb8YUQw&r=-KNeEZI7ShpbkRX5FgA_3aDmyJJkbPigBmPHDUbP87k&m=YrvulSuYq3tNn-NEQY_BfrswkODcz558egaaqt7-1N_W7XlQAhfwnMQbXvltosgS&s=QuVX0AYci987M4nRfpGcFekhlzPi57Zlnav1XDYQjmE&e=>:

> +do
+if [ -z "${JYPATH}" ] ; then
+JYPATH="$i"
+else
+JYPATH=${JYPATH}:"$i"
+fi
+done
+#-------------------------------------------------------------
+echo " "
+echo " "
+echo "*****************************************"
+echo "*    Running COAT-JAVA JShell Scripts   *"
+echo "*****************************************"
+echo " "
+echo " "
+jshell --class-path "$JYPATH" "$@"


Do you need to export JAVA_OPTS too, as bin/run-groovy does?

On that note, this script has significant overlap with bin/run-groovy, which makes me wonder if they could be combined (e.g., move common things like JYPATH into bin/env.sh... any thoughts @baltzell<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_baltzell&d=DwMCaQ&c=CJqEzB1piLOyyvZjb8YUQw&r=-KNeEZI7ShpbkRX5FgA_3aDmyJJkbPigBmPHDUbP87k&m=YrvulSuYq3tNn-NEQY_BfrswkODcz558egaaqt7-1N_W7XlQAhfwnMQbXvltosgS&s=6QrI7y0MjGXupuSp6EAhGPhmit51Cg20fCGCd6Mfa64&e=>?

File: bin/run-jshell
Comment (c-dilks) [reply bin/run-jshell:50]: From some discussion today, we are fine with the repetition.

**Review by c-dilks — APPROVED** [review]

**Review by baltzell — APPROVED** [review]

**Pull request merged**
